src/scenes/TradeHub/index.js

```diff
@@ -82,141 +69,160 @@ class TradeHub extends Component {
       selectedCrypto: crypto
-    },() => this.rebindSocket());
-  }
+    }, () => this.rebindSocket());
+  };
+
+  checkPrice = (price) => {
+    if (price > this.state.boughtPrice) return;
+
+    if (this.isHighestPrice(price)) {
+      this.setHighestPrice(price);
+      return;
+    }
+
+    if (this.shouldSell(price)) this.sell(price);
+  };
+
+  setBoughtPrice = price => this.setState({ boughtPrice: price });
 
-  render() {
-    const selectedCrypto = this.state.selectedCrypto;
+  setDiffPercentage = percentage =>  this.setState({ diffPercentage: percentage / 100 });
+
+  setSellEnabled = value => this.setState({ sellEnabled: value });
+
+  isHighestPrice = price => price > this.state.highestPrice;
+
+  setHighestPrice = price => this.setState({ highestPrice: price });
+
+  shouldSell = (price) => {
+    const { sellEnabled, hasSold, highestPrice, diffPercentage } = this.state;
+    if (sellEnabled && !hasSold) {
+      return price <= highestPrice - highestPrice * diffPercentage;
+    }
+  };
+
+  sell = (price) => {
+    alert('SOLD at: ' + price);
+    this.setState({
+      hasSold: true
+    });
+  };
+
+  render = () => {
+    const { sellEnabled, selectedCrypto, cryptoList, boughtPrice } = this.state;
+    const diffPercentage = this.state.diffPercentage * 100;
     const currentPrice = this.state.currentPrice.toFixed(6);
     const highestPrice = this.state.highestPrice.toFixed(6);
     const sellPrice = (highestPrice - highestPrice * this.state.diffPercentage).toFixed(6);
-    const cryptoList = this.state.cryptoList;
+
+    const sellOptions = [
+      {
+        label: 'Enable',
+        value: true,
+        color: 'success'
+      },
+      {
+        label: 'Disable',
+        value: false,
+        color: 'danger'
+      }
+    ];
 
     return (
       <div>
         <Header />
 
-        <div className="container">
+        <Container>
           <HorizontalTabList list={cryptoList} selectedItem={this.state.selectedCrypto} changeSelected={this.changeSelectedCrypto}/>
-          <div className="row">
-            <div className="col-sm"><InfoPanel title={selectedCrypto.quoteAsset + " " + currentPrice} description={selectedCrypto.baseAsset + " current price"}/></div>
-            <div className="col-sm"><InfoPanel title={selectedCrypto.quoteAsset + " "  + highestPrice} description={selectedCrypto.baseAsset + " hightest price since bought"}/></div>
-            <div className="col-sm"><InfoPanel title={selectedCrypto.quoteAsset + " "  + sellPrice} description={selectedCrypto.baseAsset + " price to sell on"}/></div>
-          </div>
-        </div>
-
-        <div className="container mt-3">
-          <div className="row">
-            <div className="col-sm"><Chart selectedCrypto={this.state.selectedCrypto} /></div>
-          </div>
-        </div>
+          <Row>
+            <Col>
+              <InputPanel
+                name='bought_price'
+                value={boughtPrice}
+                step='0.01'
```

Comment:
It'd be preferred if you'd use a number for `step` but I don't know if that's possible

Reply:
Step gets passed into a input field within InputPanel. The input field expects a string as step type.